gomock/call.go

```diff
@@ -156,6 +156,9 @@ func (c *Call) isPreReq(other *Call) bool {
 
 // After declares that the call may only match after preReq has been exhausted.
 func (c *Call) After(preReq *Call) *Call {
+	if c == preReq {
+		panic("a call isn't allowed to be it's own prerequisite")
```

Comment:
Good idea.

Here I simply followed the existing pattern (using a panic like a few lines below). I'm going to switch to `c.t.Fatal` and adjust the tests.